app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/CustomSelectorActivity.kt

```diff
@@ -112,6 +141,82 @@ class CustomSelectorActivity: BaseActivity(), FolderClickListener, ImageSelectLi
     private fun setUpBottomLayout() {
         val done : Button = findViewById(R.id.upload)
         done.setOnClickListener { onDone() }
+
+        val notForUpload : Button = findViewById(R.id.not_for_upload)
+        notForUpload.setOnClickListener{ onClickNotForUpload() }
+    }
+
+    /**
+     * Gets selected images and proceed for database operations
+     */
+    private fun onClickNotForUpload() {
+        val selectedImages = viewModel.selectedImages.value
+        if(selectedImages.isNullOrEmpty()) {
+            markAsNotForUpload(arrayListOf())
+            return
+        }
+        var i = 0
+        while (i < selectedImages.size) {
+            val path = selectedImages[i].path
+            val file = File(path)
+            if (!file.exists()) {
+                selectedImages.removeAt(i)
+                i--
+            }
+            i++
+        }
+        markAsNotForUpload(selectedImages)
+    }
+
+    /**
+     * Insert selected images in the database
+     */
+    private fun markAsNotForUpload(images: ArrayList<Image>) {
+        insertIntoNotForUpload(images)
+    }
+
+    /**
+     * Initializing ImageFragment
+     */
+    fun setOnDataListener(imageFragment: ImageFragment?) {
+        this.imageFragment = imageFragment
+    }
+
+    /**
+     * Insert images into not for upload
+     */
+    fun insertIntoNotForUpload(images: java.util.ArrayList<Image>) {
+        scope.launch {
+            images.forEach {
+                val imageSHA1 = getImageSHA1(it.uri)
+                notForUploadStatusDao.insert(
+                    NotForUploadStatus(
+                        imageSHA1,
+                        true
+                    )
+                )
+            }
+            imageFragment!!.refresh()
```

Comment:
@4D17Y4 I have to include the refresh function inside the scope. Otherwise, UI will be changed before database operations. I was thinking to change the name of the function. But can't finalize an appropriate one.

Reply:
Ohk, take note of it so that you dont forget. Better if you add a comment here, just as a reminder.

Reply:
Can you suggest an appropriate name for the function?